woocommerce/woocommerce-android
# Handle attributes disabled for variations usage

Summary
==========
Fixes issues #3771, #3770, #3913 and #3769 by creating filters to only enable for Variations attributes set as true for the `use for variation` flag. Since issue #3770 is not a bug from the API, but a consequence of trying to update inside a Variation attributes who aren't enabled for variation usage, this PR won't solve the issue itself but block the described scenario from happening.

How to test
==========
Before testing these changes, make sure to set up a Variable Product with Attributes flagged as both `true` and `false` for `use for variations`, so we can properly reproduce a scenario with Attributes to be filtered out.

Make sure that for every Variation view, only the enabled for variation usage are displayed (except the Add Attributes view, since that would be view where we want to add Attributes who are disabled for variations)

![image](https://user-images.githubusercontent.com/5920403/116168413-ae28fa80-a6d8-11eb-878d-ceb13c4e0071.png)

# Before
| Product Detail | Variation List | Variation Detail | Edit Variation Attributes | Attributes List |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| ![Screenshot_20210426_220719](https://user-images.githubusercontent.com/5920403/116169794-e67e0800-a6db-11eb-8400-1e86116e096d.png) | ![Screenshot_20210426_220723](https://user-images.githubusercontent.com/5920403/116169803-ec73e900-a6db-11eb-8fe0-97b77e05fd50.png) | ![Screenshot_20210426_220727](https://user-images.githubusercontent.com/5920403/116169808-ee3dac80-a6db-11eb-9344-3e4cbe7f1537.png) | ![Screenshot_20210426_220738](https://user-images.githubusercontent.com/5920403/116169818-f1389d00-a6db-11eb-8ef4-69f6ffa82ae7.png) | ![Screenshot_20210426_220749](https://user-images.githubusercontent.com/5920403/116169823-f269ca00-a6db-11eb-8828-4fe757f255f4.png) |

# After
| Product Detail | Variation List | Variation Detail | Edit Variation Attributes | Attributes List |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| ![Screenshot_20210426_215900](https://user-images.githubusercontent.com/5920403/116169492-2a244200-a6db-11eb-8355-677110cb1099.png) | ![Screenshot_20210426_215845](https://user-images.githubusercontent.com/5920403/116169495-2bee0580-a6db-11eb-9db9-22fe8625eb7f.png) | ![Screenshot_20210426_215851](https://user-images.githubusercontent.com/5920403/116169498-2d1f3280-a6db-11eb-8cb3-4529256f50bc.png) | ![Screenshot_20210426_215854](https://user-images.githubusercontent.com/5920403/116169504-2f818c80-a6db-11eb-85f4-3bd782c2bd2d.png) | ![Screenshot_20210426_221049](https://user-images.githubusercontent.com/5920403/116169990-496f9f00-a6dc-11eb-869c-3f045e4442d2.png) |

Update release notes:
